src/components/shared/Logo/UnitLogo.js

```diff
@@ -14,28 +14,45 @@ import portraitML from '../../../images/portrait_ml.png';
 const portraits = { bs: portraitBS, rb: portraitRB, ml: portraitML };
 
 const UnitLogo = props => {
-  const { unitId = null, collectionId = null, sourceId = null, width = 120, className = '', fallbackImg = 'default', ...rest } = props;
+  const {
+    unitId       = null,
+    collectionId = null,
+    sourceId     = null,
+    width        = 120,
+    height,
+    className    = '',
+    fallbackImg  = 'default',
+    ...rest
+  }          = props;
   const sourcePath = useSelector(state => sources.getPathByID(state.sources)(sourceId));
 
-  const src = unitId !== null ? Requests.imaginary('thumbnail', {
-    url: assetUrl(`api/thumbnail/${unitId}`),
-    width,
-    stripmeta: true,
-  }) : null;
+  let src;
+  if (unitId) {
+    src = Requests.imaginary('thumbnail', {
+      url: assetUrl(`api/thumbnail/${unitId}`),
+      width,
+      stripmeta: true,
+    });
+  } else if (collectionId) {
+    src = Requests.imaginary('thumbnail', {
+      url: assetUrl(`logos/collections/${collectionId}.jpg`),
+      width,
+      height,
+      stripmeta: true
+    });
+  }
```

Comment:
Maybe you need to add another else with src = null
Maybe also move this block to a function

Reply:
fixed